devices/ak2i.cpp

```diff
@@ -61,30 +67,52 @@ class AK2i : Flashcart {
         cmdbuf[2] = (address >>  8) & 0xFF;
         cmdbuf[3] = (address >>  0) & 0xFF;
 
-        sendCommand(cmdbuf, 0, nullptr, (m_ak2i_hwrevision == 0x81818181) ? 20 : 0 );
+        sendCommand(cmdbuf, 0, nullptr, (m_ak2i_hwrevision == 0x81) ? 20 : 0 );
         a2ki_wait_flash_busy();
     }
 
-    void a2ki_writebyte(uint32_t address, uint8_t value) {
+    uint32_t rawErase(uint32_t address) {
+        return erase_wrapper<AK2i, &AK2i::ak2i_erase, page_size>(address);
+    }
+
+    uint32_t rawWrite(uint32_t address, uint32_t length, const uint8_t *buffer) {
         uint8_t cmdbuf[8];
+        (void)length; // Unused, since we only write one byte at a time.
 
-        if (m_ak2i_hwrevision == 0x44444444)
+        if (m_ak2i_hwrevision == 0x44)
         {
             memcpy(cmdbuf, ak2i_cmdWriteByteFlash, 8);
             cmdbuf[1] = (address >> 16) & 0x1F;
         }
-        else if (m_ak2i_hwrevision == 0x81818181)
+        else if (m_ak2i_hwrevision == 0x81)
         {
             memcpy(cmdbuf, ak2i_cmdWriteByteFlash81, 8);
             cmdbuf[1] = (address >> 16) & 0xFF;
         }
 
         cmdbuf[2] = (address >>  8) & 0xFF;
         cmdbuf[3] = (address >>  0) & 0xFF;
-        cmdbuf[4] = value;
+        cmdbuf[4] = *buffer;
 
         sendCommand(cmdbuf, 0, nullptr, 20);
         a2ki_wait_flash_busy();
+
+        return 1;
+    }
+
+    void setWriteState(bool status) {
```

Comment:
this is a much better way to handle the state changes

Reply:
mm, I may want to bring it over into DSTT land as well.